aten/src/ATen/native/SparseMM.cpp

```diff
@@ -0,0 +1,85 @@
+#include "ATen/ATen.h"
+#include "ATen/Dispatch.h"
+#include "ATen/NativeFunctions.h"
+
+#include <TH/TH.h>
+#include <THNN/THNN.h>
+#undef THNN_
+#include <THS/THS.h>
+
+#include "ATen/CPUFloatTensor.h"
+#include "ATen/CPUDoubleTensor.h"
+#include "ATen/SparseCPUFloatTensor.h"
+#include "ATen/SparseCPUDoubleTensor.h"
+
+namespace at {
+namespace native {
+
+// Calling into TH for sspaddmm because ATen code generation currently
+// doesn't support Sparse x Dense operations on Sparse tensors
+template <class scalar>
+void TH_sspaddmm(Tensor & result, Scalar beta, const Tensor& self,
+    Scalar alpha, const Tensor& mat1, const Tensor& mat2) {
+  runtime_error("sspaddmm NYI for passed in types\n");
+}
+
+template <>
+void TH_sspaddmm<float>(Tensor& result, Scalar beta, const Tensor& self,
```

Comment:
I don't think you should call this TH_; this function just calls into TH like most other ATen functions (which aren't prefixed with TH).  It's fine to call it something like sspaddmm_TH_dispatch or something.